plugins/discourse-ai/spec/lib/modules/automation/llm_triage_spec.rb

```diff
@@ -142,11 +143,100 @@ def triage(**args)
 
     reviewable = ReviewablePost.last
 
-    expect(reviewable.target).to eq(post)
+    expect(reviewable.target_id).to eq(post.id)
+    expect(reviewable.target_type).to eq("Post")
     expect(reviewable.reviewable_scores.first.reason).to include("bad")
     expect(post.reload).to be_hidden
   end
 
+  it "can handle flag + delete" do
+    DiscourseAi::Completions::Llm.with_prepared_responses(["bad"]) do
+      triage(
+        post: post,
+        triage_persona_id: ai_persona.id,
+        search_for_text: "bad",
+        flag_post: true,
+        flag_type: :review_delete,
+        automation: nil,
+      )
+    end
+
+    reviewable = ReviewablePost.last
+
+    expect(reviewable.target_id).to eq(post.id)
+    expect(reviewable.target_type).to eq("Post")
+    expect(reviewable.reviewable_scores.first.reason).to include("bad")
+    expect(post.reload.trashed?).to eq(true)
+  end
+
+  it "restores deleted post when moderator approves" do
+    DiscourseAi::Completions::Llm.with_prepared_responses(["bad"]) do
+      triage(
+        post: post,
+        triage_persona_id: ai_persona.id,
+        search_for_text: "bad",
+        flag_post: true,
+        flag_type: :review_delete,
+        automation: nil,
+      )
+    end
+
+    reviewable = ReviewablePost.last
+    expect(post.reload.trashed?).to eq(true)
```

Comment:
```suggestion
    expect(post.reload.trashed?).to eq(true)
    expect(post.topic.reload.trashed?).to eq(true)
```

Later we are checking that topic was restored so let's ensure that topic was soft deleted as well